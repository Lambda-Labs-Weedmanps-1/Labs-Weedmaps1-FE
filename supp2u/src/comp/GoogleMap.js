import React from 'react'
import { GoogleMap, Marker, withGoogleMap,withScriptjs } from "react-google-maps"


const GoogleMaps = withScriptjs(withGoogleMap((props) =>

  <GoogleMap
  className="map"
    defaultZoom={8}
    defaultCenter={{ lat: 23.118813, lng: -82.329933  }}
  >
    {props.isMarkerShown && <Marker position={{ lat: 23.118813, lng: -82.329933 }} />}
  </GoogleMap>))


export default GoogleMaps;