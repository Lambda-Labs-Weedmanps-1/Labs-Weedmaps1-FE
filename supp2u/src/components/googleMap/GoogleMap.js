import React, {createRef,} from 'react';
import "./style.sass"
import { mapStyle } from "./MapStyles";
import axios from "axios"

class GoogleMaps extends React.Component{

  mapRef = createRef();
  // googleMapDiv = ()=> {return document.getElementById("google-map")};
  createGoogleMap = () =>{
    let map = new window.google.maps.Map(this.mapRef.current, {
      zoom: 16,
      center:{
        lat: 23.118813,
        lng: -82.329933
      },
      styles: mapStyle
    });
    this.creatMarker(map);
  };
  creatMarker =(map)=>{
    let marker = new window.google.maps.Marker({
      position: {
        lat: 23.118813,
        lng: -82.329933
      },
      map: map,
    });
    marker.addListener("click", props=> console.log("marker clicked", props));
  };
  componentDidMount() {
    const googleMapScript = document.createElement('script');
    googleMapScript.src = `https://maps.googleapis.com/maps/api/js?key=${process.env.REACT_APP_GOOGLE_MAP_API_KEY}&v=3.exp&libraries=geometry,drawing,places`;
    window.document.body.appendChild(googleMapScript);
    googleMapScript.addEventListener("load", ()=>{
      this.googleMap = this.createGoogleMap();
    })
  
  }

  render(){
    return <div id={"google-map"} className={"map"} ref={this.mapRef}/>
  }
}

export default GoogleMaps;