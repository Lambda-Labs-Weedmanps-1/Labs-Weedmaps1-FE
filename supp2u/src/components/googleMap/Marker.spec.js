import React from 'react';
import GoogleMap from './GoogleMap';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

//This test makes sure that marker isn't crashing the app, backend testing will make sure we have positions for the marker
describe('Marker has positions', ()=>{
    it ('Marker function loads without crashing Google Maps', ()=>{
        const GMap = shallow(<GoogleMap />);
    
      expect(GMap).toMatchSnapshot();
    });
})