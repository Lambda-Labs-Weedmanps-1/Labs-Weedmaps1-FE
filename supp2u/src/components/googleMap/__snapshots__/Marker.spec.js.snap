// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Marker has positions Marker function loads without crashing Google Maps 1`] = `
<div
  className="map"
  id="google-map"
/>
`;
